### PR TITLE
Added readthedocs support for documentation

### PR DESCRIPTION
**Pull request by michaeldboyd**
First, I recognize that this looks like a large PR, but the changes I've made were pretty simple. 

As discussed in the HIPE [0025-indy-docs-framework](https://github.com/hyperledger/indy-hipe/blob/master/text/0025-indy-docs-framework/README.md) we would like to have a centralized site for our documentation. The changes I have made are outlined in that HIPE. 

Here, you will see a couple of things: 
- I've changed the `doc/` directory to the `docs/` directory to match the rest of the indy repositories. This change was helpful for the automated docs builder. 
- Fixed links to the old `doc/` directory within the READMEs 
- Organized the `docs/` directory into subdirectories based on topic
- Deleted the `/doc/dkms/DKMS Design and Architecture V3.md` because it was a stub file that only referenced the newer [doc here](https://github.com/hyperledger/indy-sdk/blob/677a0439487a1b7ce64c2e62671ed3e0079cc11f/doc/design/005-dkms/DKMS%20Design%20and%20Architecture%20V3.md)
- Most of the files changed are just name changes because of the directory rename. 

The significant files added are
- `conf.py` - this must be added to the root directory to include `docs/` folders from subproject like vcx.
- `docs/README.md`

Thanks for taking the time to review!

### CODE REVIEW

**Comment by michaeldboyd** [comment]
@Artemkaaas @vimmerru Can someone please review this PR to be merged?

File: docs/getting-started/indy-walkthrough.md [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
-      * [Step 2: Connecting to the Indy Nodes Pool](#step-2-connecting-to-the-indy-nodes-pool)
-      * [Step 3: Getting the Ownership for Stewards's Verinym](#step-3-getting-the-ownership-for-stewardss-verinym)
-      * [Step 4: Onboarding Faber, Acme, Thrift and Government by the Steward](#step-4-onboarding-faber-acme-thrift-and-government-by-steward)
-        * [Connecting the Establishment](#connecting-the-establishment)
-        * [Getting the Verinym](#getting-verinym)
-      * [Step 5: Credential Schemas Setup](#step-5-credential-schemas-setup)
-      * [Step 6: Credential Definition Setup](#step-6-credential-definition-setup)
-  * [Alice Gets a Transcript](#alice-gets-a-transcript)
-  * [Apply for a Job](#apply-for-a-job)
-  * [Apply for a Loan](#apply-for-a-loan)
```
Comment (jovfer) [inline docs/getting-started/indy-walkthrough.md]: Is it intentional to remove last two items from the list?

**Comment by cam-parra** [comment]
@michaeldboyd Please review the concern above.

**Comment by michaeldboyd** [comment]
@cam-parra @jovfer I have resolved all conflicts. Are there any other blocking changed I need to make?

**Comment by mattraffel** [comment]
If you can resolve the merge conflict, we should be good.  I've started a CI build since it is required.

**Comment by michaeldboyd** [comment]
Done

**Pull request merged**